hasjob/templates/macros.html.jinja2

```diff
@@ -216,3 +237,71 @@
+       'maxValue' - max value of conversion funnel across job posts of last 30 days
+       'firstColour' & 'secondColour' - object containing offset & colour code(in hex)
+        {
+          "offset": 0.8
+          "colour": "#E05F26"
+        }
+        These are colours to define the colour gradient canvas.
+      */
+      createGradientColourBar: function(funnelName, maxValue, firstColour, secondColour){
+        var canvas = document.createElement("canvas");
+        canvas.id = funnelName;
+        canvas.width = maxValue;
+        canvas.height = 10;
+
+        var context = canvas.getContext('2d');
+        context.rect(0, 0, canvas.width, canvas.height);
+        var grd = context.createLinearGradient(0, 0, canvas.width, canvas.height);
+
+        grd.addColorStop(firstColour.offset, firstColour.colour);
+        grd.addColorStop(secondColour.offset, secondColour.colour);
+
+        context.fillStyle = grd;
+        context.fill();
+        //Store the canvas context and end colour of the conversion funnel, later to be used by setFunnelColour for picking the colour for a conversion funnel value for a job post.
+        window.Hasjob.Config[funnelName] = {};
+        window.Hasjob.Config[funnelName].canvasContext = context;
+        window.Hasjob.Config[funnelName].maxColour = secondColour.colour;
+      },
+      /* 
+        Picks the colour for the value from the colour gradient canvas based on a scale of 0 to maxValue.
+        Takes funnelName', 'value, elementId'
+       'funnelName' - conversion funnel's name. 
+       'value' - conversion funnel value for the job post
+       'elementId' - id attribute of the element of which background colour is to be set
+      */
+      setFunnelColour: function(funnelName, value, elementId){
+        //rgba - RGBA values at a particular point in the canvas.
+        var rgba = window.Hasjob.Config[funnelName].canvasContext.getImageData(value, 1, 1, 1).data;
+        if (rgba[0] > 255 || rgba[1] > 255 || rgba[2] > 255) {
+          // rgb value is invalid, hence return white
+          colourHex ="#FFFFFF";
+        } else if (rgba[0] == 0 && rgba[1] == 0 && rgba[2] == 0) {
+          // value greater than maxValue hence return the last colour of the gradient
+          colourHex = window.Hasjob.Config[funnelName].maxColour;
+        } else {
+          // Get the colour code in hex from RGB values returned by getImageData
+          colourHex = "#" + (("000000" + (rgba[0] << 16) | (rgba[1] << 8) | rgba[2]).toString(16)).slice(-6);
+        }
+        // Set the background colour of the element
+        document.getElementById(elementId).style.backgroundColor = colourHex;
+      }
+    }
+  </script>
+{%- endmacro -%}
+
+{%- macro set_funnel_colour(funnelName, value, elementId) -%}
+  <script type="text/javascript">
+    window.Hasjob.FunnelStat.setFunnelColour("{{funnelName}}", {{value}}, "{{elementId}}");
+  </script>
```

Comment:
Not a wise idea to insert four script tags for each stickie. Better to insert `data-` elements and parse through them at once.

Reply:
Fixed. Moved it to app.js